store/retry.go

```diff
@@ -20,12 +20,45 @@
 package store
 
 import (
-	"gopkg.in/retry.v1"
+	"fmt"
 	"io"
 	"net"
 	"net/http"
+	"time"
+
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/osutil"
+	"gopkg.in/retry.v1"
 )
 
+// the LimitTime should be slightly more than 3 times of our http.Client
+// Timeout value
+var defaultRetryStrategy = retry.LimitCount(5, retry.LimitTime(33*time.Second,
+	retry.Exponential{
+		Initial: 100 * time.Millisecond,
+		Factor:  2.5,
+	},
+))
+
+func maybeLogRetryAttempt(url string, attempt *retry.Attempt, startTime time.Time) {
+	if osutil.GetenvBool("SNAPPY_TESTING") || attempt.Count() > 1 {
+		delta := time.Since(startTime) / time.Millisecond
+		logger.Debugf("Retyring %s, attempt %d, delta time=%v ms", url, attempt.Count(), delta)
```

Comment:
remember you can just pass a `time.Duration` to `%s` and it'll print e.g. `123ms` if that is the most appropriate unit for the duration.